go/mysql/datetime/types.go

```diff
@@ -212,6 +245,69 @@ func (d Date) SundayWeek() (int, int) {
 	return sun.Year(), (sun.YearDay()-1)/7 + 1
 }
 
+const DefaultWeekMode = 0
+
+func (d Date) Week(mode int) int {
+	switch mode {
+	case 0:
+		year, week := d.SundayWeek()
+		if year < d.Year() {
+			return 0
+		}
+		return week
+	case 1:
+		year, week := d.ISOWeek()
+		if year < d.Year() {
+			return 0
+		}
+		return week
+	case 2:
+		_, week := d.SundayWeek()
+		return week
+	case 3:
+		_, week := d.ISOWeek()
+		return week
+	case 4, 5, 6, 7:
+		// TODO
+		return 0
```

Comment:
This (and YearWeek) feels like we're explicitly allowing for unexpected behavior. I do see that you already have a separate PR open for this though.

Reply:
@mattlord Yeah, looking into this in a follow up to fix these and add support for them.

Reply:
@mattlord See https://github.com/vitessio/vitess/pull/12916